samcli/commands/delete/delete_context.py

```diff
@@ -57,9 +58,15 @@ def __enter__(self):
         self.parse_config_file()
         if not self.stack_name:
             LOG.debug("No stack-name input found")
-            self.stack_name = prompt(
-                click.style("\tEnter stack name you want to delete:", bold=True), type=click.STRING
-            )
+            if not self.no_prompts:
+                self.stack_name = prompt(
+                    click.style("\tEnter stack name you want to delete:", bold=True), type=click.STRING
+                )
+            else:
+                raise click.BadOptionUsage(
+                    option_name="delete",
```

Comment:
`option_name` should match the option being used. In this case, it should be `"--stack-name"`